vim-patch:9.0.{0114,0115,0118}

Shougo opened the pull request:
https://github.com/vim/vim/commit/f39cfb72629f3e7fefaf578a3faa2619cd0654f8

https://github.com/vim/vim/commit/6747cf1671bd41cddee77c65b3f9a70509f968db

Shougo committed:
vim-patch:9.0.0114: the command line takes up space even when not used

Problem:    The command line takes up space even when not used.
Solution:   Allow for 'cmdheight' to be set to zero. (Shougo Matsushita,
            closes vim/vim#10675, closes vim/vim#940)
https://github.com/vim/vim/commit/f39cfb72629f3e7fefaf578a3faa2619cd0654f8

Omit win_redr_ruler() change: winbar may still need redraw.
Omit win_update() changes: Nvim doesn't use `Rows` there.
Omit redraw_asap(): removed.

Shougo committed:
vim-patch:9.0.0115: when 'cmdheight' is zero pressing ':' may scroll a window

Problem:    When 'cmdheight' is zero pressing ':' may scroll a window.
Solution:   Add the made_cmdheight_nonzero flag and set 'scrolloff' to zero.
https://github.com/vim/vim/commit/6747cf1671bd41cddee77c65b3f9a70509f968db

zeertzjq commented on src/nvim/screen.c:
```diff
@@ -6501,8 +6501,8 @@ static void win_redr_ruler(win_T *wp, bool always)
-  // If 'ruler' off or redrawing disabled, don't do anything
-  if (!p_ru) {
+  // If 'ruler' off or messages area disabled, don't do anything
+  if (!p_ru || (wp->w_status_height == 0 && p_ch == 0))
```
This is probably needed for winbar.

Shougo replied:
OH

Shougo replied:
Fixed.

zeertzjq committed:
vim-patch:9.0.0118: no test for what patch 9.0.0155 fixes

Problem:    No test for what patch 9.0.0155 fixes.
Solution:   Add a test. Fix typos.  (closes vim/vim#10822)
https://github.com/vim/vim/commit/750209459c9e54030409afe8f4ad59570600b5c4
